fix: lupogg king earthquake death errors and room death messages

adrienlaws opened the pull request:
## What
- Fixes #1233 
- Fixed `DoEarthQuake` in `earthqua.kod` to properly handle two modes:
   - Caster mode (`#who`): distance-based damage falloff, deaths blamed on caster
   - Environmental mode (`#where`): flat damage, deaths blamed on room
   - Added `bEnvironmental` flag to make mode handling clearer
   - Added debug warning if both `#who` and `#where` are passed
   - Parameters: `rDeathSourceDef`, `rDeathSourceName` for pre-captured flavor message strings
- Added new `earthquake_death_rsc` for monster death cast (replaces generic `earthquake_cast_rsc`)
- Changed lupogg king death earthquake in `lupking.kod` to use room-based mode with `Post` for proper timing
- Fixed room death message in `system.kod`:
   - Adjusted `system_user_killed_room` message for environmental deaths
   - Fixed murderer/outlaw status from overwriting room death messages

## Why

- When the lupogg king died, its earthquake caused server errors:
  ```
  [earthqua.bof] InterpretBinaryAssign can't send MESSAGE SquaredDistanceTo to non-object 0,0
  ```
- The error occurred because `ComputeDamage` tried to calculate distance from the dying/deleted lupogg king object
- Players killed by the earthquake saw "### PlayerName was just slain by their own folly" (suicide message) because the server error corrupted the killer parameter
- Room deaths had no global broadcast message - players would die silently to environmental hazards

## How

- Lupogg king pre-captures its name/article strings *before* calling `Post`:
  ```kod
  rMyDef = Send(self,@GetDef);
  rMyName = Send(self,@GetName);
  Post(oSpell,@DoEarthQuake,#where=poOwner,#severity=2,
       #rDeathSourceDef=rMyDef,#rDeathSourceName=rMyName);
  ```
- Environmental earthquakes use flat damage (`Random(5,9) * severity`) instead of distance-based
- Environmental earthquakes blame the room for kills, triggering the room death message
- Room death message is simple and ominous:
  ```kod
  system_user_killed_room = "### %q met an untimely end."
  ```
- Murderer/outlaw message overrides are skipped when killer is a Room (they require player killer params)

### Flow Diagram

```mermaid
flowchart TD
    subgraph KILLED["lupking.kod Killed()"]
        K1[Lupogg King dies] --> K2["Pre-capture: rMyDef, rMyName"]
        K2 --> K3["Post earthquake with #where=poOwner"]
        K3 --> K4[propagate - king cleanup begins]
    end
    
    subgraph POST["Posted Earthquake Fires Later"]
        E1["DoEarthQuake receives #where"] --> E2[bEnvironmental = TRUE]
        E2 --> E3[Flat damage, room blamed]
        E3 --> E4[Flavor message uses pre-captured strings]
    end
    
    K4 --> |"King object deleted"| POST
    
    style K2 fill:#2f9e44,color:#fff
    style E3 fill:#2f9e44,color:#fff
    style E4 fill:#2f9e44,color:#fff
```

### Earthquake Modes

| Parameter | Mode | Damage | Kill Blame | Use Case |
|-----------|------|--------|------------|----------|
| `#who` | Caster | Distance-based falloff | Caster | Player/monster casting spell |
| `#where` | Environmental | Flat (severity * 5-9) | Room | Boss death, environmental hazard |

## Example

### Before (Broken)
- Server errored on lupogg king death
- No global death message for environmental kills was presented
  - Previously used "slain by their own folly" (wrong)

### After (Fixed)
- No server errors
- Local message: "As the lupogg king collapses, the earth shudders with its dying fury!"
- Global death message: "### PlayerName met an untimely end."
- Proper timing: king dies first, then earthquake fires
<img alt="meridian_PoCwCmxab6" src="https://github.com/user-attachments/assets/6ee4d846-1958-4fba-b0c2-f4117128b7ea" />


## Notable
- `severity=2` means damage range is 10-18 (base 5-9 * 2)
- Boss monsters are protected from environmental earthquake damage (existing behavior preserved)
- Room death message intentionally doesn't reveal location (keeps mystery/danger)

adrienlaws commented on kod/object/active/holder/nomoveon/battler/monster/lupking.kod:
```diff
@@ -471,7 +471,7 @@ messages:
    
       oSpell = Send(SYS,@FindSpellByNum,#num=SID_EARTHQUAKE);
-      post(oSpell,@DoEarthQuake,#where=poOwner,#severity=2);
+      post(oSpell,@DoEarthQuake,#who=self,#where=poOwner,#severity=2);
```
When testing I saw no errors when switching to use `#who=self` and keeping the `#where=poOwner`

I think the reason the spell used `#where` is because that way the attribution of the death isn't from the Lupogg King.  But as it stands it ends up giving the messaged "slain by their own folly".  I see comments about the "killer" here in [earthqua.kod](https://github.com/Meridian59/Meridian59/blob/207374e64d09e9a65071e9beb98bc557023a281c/kod/object/passive/spell/earthqua.kod#L122)

So after this change the message is:

>  The notorious murderer, test2, has been killed by a lupogg king.

If you are killed by the subsequent earthquake.  To me this makes sense. At least more than "killed by their own folly."  Maybe the original goal was to have the room kill them and it would use the room death message of
>  The sewers have laid claim to yet another poor soul, test2.

But room death messages seem broken so it uses the folly message instead.

I removed the where parameter and kept who, and under testing it seems normal.

akirmse replied:
This means that calling DoEarthquake with #where is broken, and since we're not fixing it here, it should be removed.  So please combine CastSpell and DoEarthquake and remove the where parameter.  In LupoggKing, you'll instead call CastSpell with the appropriate iSpellPower (25 I guess?)

I'm pretty concerned about calling Send instead of Post here.  Usually Post is called for a reason, to change the order of operations.  If you Send instead, then the earthquake can kill people before the lupogg king is done dying.  That will reverse the order of death messages sent out, which maybe isn't that big a deal, but it could have other consequences that we don't understand.  You're sure it's OK?

adrienlaws commented:
Refactored this PR and updated the description.
- 1/ fixed, instead of removed, where / environmental earthquake and the related $/NIL errors and incorrect death message from lupogg king death earthquake 
> slain by their own folly
- 2/ fixed room death messages
> met an untimely end

adrienlaws commented on kod/object/passive/spell/earthqua.kod:
```diff
@@ -119,69 +121,82 @@ messages:
+                rDeathSourceDef=$, rDeathSourceName=$)
+   "Deals damage to all battlers in a room with earthquake effects.  Caster "
+   "mode (#who) applies distance-based damage falloff; environmental mode "
+   "(#where) applies flat damage.  Pre-captured rDeathSourceName/Def strings "
```
Adjusted

adrienlaws commented on kod/object/passive/spell/earthqua.kod:
```diff
@@ -198,18 +213,36 @@ messages:
             {
-               if each_obj = who
+               % Death-triggered earthquake (e.g. lupogg king dying)
+               % Use pre-captured strings since the source object may be deleted.
```
Simplified
